pkg/cluster/registerWithHive.go

```diff
@@ -0,0 +1,127 @@
+
+	m.log.Print("collecting registration data from new cluster")
+
+	cv, err := m.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	clusterid := cv.Spec.ClusterID
+
+	servicePrincipal := azure.Credentials{
+		TenantID:       m.env.TenantID(),
+		SubscriptionID: m.env.SubscriptionID(),
+		ClientID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID,
+		ClientSecret:   string(m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret),
+	}
+
+	spBytes, err := json.Marshal(servicePrincipal)
+	if err != nil {
+		return err
+	}
+
+	// TODO: What if a customer adds/modifies a pullsecret after the install ?
+	// TODO: All parameter locations to be verified
+	m.hiveWorkloadCluster = &hive.WorkloadCluster{
+		Subscription:                m.subscriptionDoc.ID,
+		Name:                        m.doc.OpenShiftCluster.Name,
+		Location:                    m.doc.OpenShiftCluster.Location,
+		BaseDomain:                  m.env.Domain(),
+		BaseDomainResourceGroupName: m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID,
+		InfraID:                     m.doc.OpenShiftCluster.Properties.InfraID,
+		ClusterID:                   string(clusterid),
+		KubeConfig:                  string(m.doc.OpenShiftCluster.Properties.AdminKubeconfig),
+		PullSecret:                  string(m.doc.OpenShiftCluster.Properties.ClusterProfile.PullSecret),
+		ServicePrincipal:            string(spBytes),
+	}
+
+	return nil
+
+}
+
+func (m *manager) initializeHiveClusterClients(ctx context.Context) error {
+	// don't fail the installation on hive related errors
+	kubeConfigPath := os.Getenv("HIVEKUBECONFIGPATH")
+	if kubeConfigPath == "" {
+		return errors.New("missing HIVEKUBECONFIGPATH env variable")
+	}
+
+	m.hiveRegistrator = hive.NewRegistrator(m.hiveWorkloadCluster)
+	err := m.hiveRegistrator.InitializeFromFile(ctx, kubeConfigPath)
+	if err != nil {
+		m.log.Print(err)
+		return err
+	}
+	return nil
+
+}
+
+func (m *manager) registerWithHive(ctx context.Context) error {
+	// don't fail the installation on hive related errors
+	if m.hiveRegistrator == nil {
+		m.log.Print("No hive registrator, skipping step")
+		return nil
+	}
+	m.log.Print("registering with hive")
+
+	isRegistered, err := m.hiveRegistrator.IsRegistered(ctx)
+	if err != nil {
+		m.log.Print(err)
+		return nil
+	}
+
+	if isRegistered {
+		m.log.Print("cluster is already registered, it seems. Skipping step.")
+		return nil
+	}
+
+	err = m.hiveRegistrator.Register(ctx)
+	if err != nil {
+		m.log.Print("could not register cluster with hive ", err)
+	}
+
+	return nil
+
+}
+
+func (m *manager) verifyRegistration(ctx context.Context) (bool, error) {
+	// Never fail
+	if m.hiveRegistrator == nil {
+		m.log.Print("no  hive registrator, skipping step")
+		return true, nil
+	}
+
+	m.log.Print("Verify cluster registration")
+	time.Sleep(10 * time.Second) // give hive some time to log in to the new cluster
```

Comment:
```suggestion
```

Let's remove sleep. This way we will try immidiately (and will probably fail which is ok). Next attempt will be made after waiting for an interval. See relevant code [here](https://github.com/Azure/ARO-RP/blob/51fd449e7e6054938388f29f64f412944d31ca15/pkg/util/steps/condition.go#L55-L63).

Reply:
removed